src/SFA.DAS.AAN.Hub.Data/Repositories/MemberDataCleanupRepository.cs

```diff
@@ -0,0 +1,121 @@
+    void DeleteMemberAudit(List<Audit> memberAudits, CancellationToken cancellationToken);
+    Task DeleteMemberApprentice(Member member, CancellationToken cancellationToken);
+    Task DeleteMemberEmployer(Member member, CancellationToken cancellationToken);
+    Task UpdateMemberFutureAttendance(Member member, CancellationToken cancellationToken);
+}
+
+[ExcludeFromCodeCoverage]
+public class MemberDataCleanupRepository : IMemberDataCleanupRepository
+{
+    private readonly IAanDataContext _context;
+
+    public MemberDataCleanupRepository(IAanDataContext context)
+    {
+        _context = context;
+    }
+    public async Task<List<Member>> GetWithdrawnOrDeletedMembers()
+    {
+        var statuses = new String[] { "Withdrawn", "Deleted" };
+        var query = _context.Members
+            .Where(m => statuses.Contains(m.Status) && m.Email != m.Id.ToString() &&
+                        m.EndDate < DateTime.Today.AddDays(-14))
+            .Include(m => m.MemberPreferences)
+            .Include(m => m.MemberProfiles)
+            .Include(m => m.Notifications)
+            .Include(m => m.Audits);
+
+        return await query.ToListAsync();
+    }
+
+    public async Task<List<Member>> GetRemovedMembers()
+    {
+        var query = _context.Members
+                .Where(m => m.Status == "Removed" && m.Email != m.Id.ToString())
+                .Include(m => m.MemberPreferences)
+                .Include(m => m.MemberProfiles)
+                .Include(m => m.Notifications)
+                .Include(m => m.Audits);
+
+        return await query.ToListAsync();
+    }
+
+    public async Task UpdateMemberDetails(Member member, CancellationToken cancellationToken)
+    {
+        var existingMember = await _context.Members.Where(m => m.Id == member.Id).FirstAsync(cancellationToken);
+
+        existingMember.FirstName = "Past";
+        existingMember.LastName = "Member";
+        existingMember.Email = member.Id.ToString();
+        existingMember.OrganisationName = "";
+        existingMember.IsRegionalChair = false;
+        existingMember.LastUpdatedDate = DateTime.UtcNow;
+    }
+
+    public void DeleteMemberProfile(List<MemberProfile> memberProfiles, CancellationToken cancellationToken)
+    {
+        _context.MemberProfiles.RemoveRange(memberProfiles);
+    }
+
+    public void DeleteMemberPreference(List<MemberPreference> memberPreferences, CancellationToken cancellationToken)
+    {
+        _context.MemberPreferences.RemoveRange(memberPreferences);
+    }
+
+    public void DeleteMemberNotifications(List<Notification> memberNotifications, CancellationToken cancellationToken)
+    {
+        _context.Notifications.RemoveRange(memberNotifications);
+    }
+
+    public void DeleteMemberAudit(List<Audit> memberAudits, CancellationToken cancellationToken)
+    {
+        var auditsToRemove = memberAudits.Select(a => a).Where(x => x.Resource != "Member").ToList();
```

Comment:
I would keep all the logic in the handler so it can be tested.

Reply:
Moved the logic to the service and unit test added